src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,130 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int INITIAL_CAPACITY = 10;
+    private static final double GROW_COEFFICIENT = 1.5;
+    private int currentCapacity = INITIAL_CAPACITY;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[INITIAL_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == currentCapacity) {
+            resize();
+        }
```

Comment:
lets move `size == currentCapacity` check into `resize()` method (this will reduce number of duplicated code)